csm/cli/support_bundle/support_bundle.py

```diff
@@ -52,57 +52,26 @@ def import_provisioner_plugin():
             return None
         return provisioner
 
-    @staticmethod
-    async def fetch_host_from_salt():
-        """
-        This method fetch hostname for all nodes from Salt DB
-        :return: hostnames : List of Hostname :type: List
-        :return: node_list : : List of Node Name :type: List
-        """
-        process = SimpleProcess(
-            "salt-call pillar.get cluster:node_list --out=json")
-        _out, _err, _rc = process.run()
-        if _rc != 0:
-            Log.warn(f"Salt command failed : {_err}")
-            return None, None
-        output = JsonMessage(_out.strip()).load()
-        nodes = output.get("local", [])
-        hostnames = []
-        for each_node in nodes:
-            process = SimpleProcess(
-                f"salt-call pillar.get cluster:{each_node}:hostname --out=json")
-            _out, _err, _rc = process.run()
-            if _rc != 0:
-                Log.warn(f"Salt command failed : {_err}")
-                return None, None
-            output = JsonMessage(_out.strip()).load()
-            hostnames.append(output.get("local", ""))
-        return hostnames, nodes
-
     @staticmethod
     async def fetch_host_from_cluster():
         """
-        This Method is Backup method for Reading Cluster.sls if Salt Read Fails.
+        This Method is for reading hostnames, node_list information from csm.conf.
         :return: hostnames : List of Hostname :type: List
         :return: node_list : : List of Node Name :type: List
         """
-        Log.info("Falling back to reading cluster information from cluster.sls.")
-        cluster_file_path = Conf.get(const.CSM_GLOBAL_INDEX,
-                                     "SUPPORT_BUNDLE>cluster_file_path")
-        if not cluster_file_path or not os.path.exists(cluster_file_path):
-            repsonse_msg = {"message": (f"{cluster_file_path} not Found. \n"
-                                        f"Please check if cluster info file is correctly configured.")}
-            return Response(rc=errno.ENOENT, output=repsonse_msg), None
-        cluster_info = Yaml(cluster_file_path).load().get("cluster", {})
-        active_nodes = cluster_info.get("node_list", [])
+        Log.info("reading hostnames, node_list information from csm.conf.")
+        node_hostname_map = dict()
+        mapping_dict = dict()
+        mapping_dict = Conf.get(const.CSM_GLOBAL_INDEX, f"{const.MAINTENANCE}")
+        node_hostname_map = mapping_dict.copy()
+        node_hostname_map.pop('shutdown_cron_time')
```

Comment:
Check for empty / NULL node_hostname_map. If empty log an error.

Reply:
done @pawankumarsrivastava